### PR TITLE
Add Channel Manager in DataNode

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #27172, #25309

### CODE REVIEW

File: internal/datanode/channel_manager.go [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+	return &cm
+}
+
+func (m *ChannelManager) Submit(info *datapb.ChannelWatchInfo) error {
+	channel := info.GetVchan().GetChannelName()
+	runner := m.getOrCreateRunner(channel)
+	return runner.Enqueue(info)
+}
+
+func (c *ChannelManager) GetProgress(info *datapb.ChannelWatchInfo) *datapb.ChannelOperationProgressResponse {
```
Comment (jiaoew1991) [inline internal/datanode/channel_manager.go:75]: Do we need some locks here to ensure concurrency for reading and writing?

Comment (XuanYang-cn) [reply internal/datanode/channel_manager.go:75]: cool, we could add a lock between finishOp and GetProgress

**Comment by XuanYang-cn** [comment]
rerun ut

File: internal/datanode/channel_manager.go [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+	allOps     map[UniqueID]*opInfo // opID -> tickler
+	opsInQueue chan *datapb.ChannelWatchInfo
+	resultCh   chan *opState
+
+	closeWg   sync.WaitGroup
+	closeOnce sync.Once
+	closeCh   chan struct{}
+}
+
+func NewOpRunner(channel string, dn *DataNode, f releaseFunc, resultCh chan *opState) *opRunner {
```
Comment (xiaofan-luan) [inline internal/datanode/channel_manager.go:224]: Move ops manager to a seperate file might be better. 
But It won't be a blocking issue

File: internal/datanode/channel_manager.go [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+	var (
+		successSig = make(chan struct{}, 1)
+		waiter     sync.WaitGroup
+	)
+
+	watchTimeout := Params.DataCoordCfg.WatchTimeoutInterval.GetAsDuration(time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), watchTimeout)
+	defer cancel()
+
+	startTimer := func(wg *sync.WaitGroup) {
```
Comment (xiaofan-luan) [inline internal/datanode/channel_manager.go:327]: timer should be a single instance?
We don't need to create one goroutine for each timer.

There should be a global ticker and process tickler for each job

File: internal/datanode/channel_manager.go (hunk 2 above)
Comment (xiaofan-luan) [inline internal/datanode/channel_manager.go:224]: each runner should have  context. 

so watchWithTimer could be canceled if Op runner is deleted

File: internal/datanode/flow_graph_manager.go [hunk 4]
```diff
@@ -215,6 +221,11 @@ func (fm *flowgraphManager) exist(vchan string) bool {
 	return exist
 }
 
+func (fm *flowgraphManager) existWithOpID(vchan string, opID UniqueID) bool {
```
Comment (xiaofan-luan) [inline internal/datanode/flow_graph_manager.go:224]: why we can't use chan as opID?
is there any special reason an opID is necessary?

**Pull request merged**